frontend/src/components/DataProductTable.vue

```diff
@@ -158,11 +248,140 @@ export default {
     perPage() {
       this.updateTotalPages();
     },
+    currentPage(val){
+      this.pageInput = val
+    },
   },
   mounted() {
     this.loadSMILESDataProducts();
+    this.pageInput = this.currentPage
   },
   methods: {
+    goToPage(){
+      const p = this.pageInput
+      if (p >= 1 && p <= this.totalPages){
+        this.currentPage = p
+        this.loadSMILESDataProducts()
+      }else{
+        alert(`Page must be 1–${this.totalPages}`)
+      }
+    },
+    toggleSelectAll() {
+    this.selectAll = !this.selectAll;
+    if (this.selectAll) {
+      this.selectedItems = this.paginatedItems.map(item => item.data_product_id);
+    } else {
+      this.selectedItems = [];
+    }
+  },
+  selectAllGlobalItems() {
+    if (this.selectAllGlobal){
+      this.selectAllGlobal = false;
+      this.selectedItems = [];
+      this.selectAll = false;
+      return;
+    }
+    this.selectAllGlobal = true;
+    this.isBusy = true; 
+    
+    utils.FetchUtils.get(
+    `/smiles/${this.type}-dps?page=1&size=${this.totalRows}`
+    ).then(resp => {
+      const all = resp.data_products || [];
+      this.selectedItems = all.map(p => p.data_product_id);
+      this.isBusy = false;
+    }).catch(err => {
+      console.error("fetch‑all ids failed:", err);
+      this.isBusy = false;
+    });
+  },
+  toggleSelectItem(itemId) {
+    const index = this.selectedItems.indexOf(itemId);
+    if (index > -1) {
+      this.selectedItems.splice(index, 1);
+    } else {
+      this.selectedItems.push(itemId);
+    }
+    const allSelected = this.paginatedItems.length > 0 &&
+      this.paginatedItems.every(item => this.selectedItems.includes(item.data_product_id));
+    this.selectAll = allSelected;
+  },
+  openShareModal() {
+  if (this.selectedItems.length === 0) {
+    alert("Please select at least one data product.");
+    return;
+  }
+  this.showShareModal = true;
+  this.fetchSuggestions("", () => {});
+  },
+  closeShareModal() {
+    this.showShareModal = false;
+    this.shareTarget = "";
+  },
+  submitShare() {
+    if (!this.shareTarget) {
+      alert("Please specify a user or group.");
+      return;
+    }
+    const requestData = {
+      share_type: this.shareType,
+      share_target: this.shareTarget,
+      ...(this.selectAllGlobal
+      ? { share_all: true }
+      : { data_product_ids: this.selectedItems }
+  )      
+    };
+    fetch("/smiles/share-data-product", {
```

Comment:
I can share a data product with myself (the logged-in user) and a non-existent user. The drop-down shouldn't allow this
<img width="824" alt="Screenshot 2025-05-06 at 5 34 03 PM" src="https://github.com/user-attachments/assets/d0b61faa-31ca-4692-84c9-dc9afbc87e80" />